corehq/apps/geospatial/reports.py

```diff
@@ -125,17 +125,27 @@ class CaseGroupingReport(BaseCaseMapReport):
 
     default_rows = 1
     force_page_size = True
+    sortable = False
 
     @property
     def headers(self):
-        headers = []
-        # ToDo: populate headers
-        return headers
+        from corehq.apps.reports.datatables import (
+            DataTablesColumn,
+            DataTablesHeader,
+        )
+        return DataTablesHeader(
+            DataTablesColumn(_("Case ID"), prop_name='case_id'),
+            DataTablesColumn(_("Case Name"), prop_name='case_name'),
+            DataTablesColumn(_("Owner ID"), prop_name='owner_id'),
+            DataTablesColumn(_("Owner Name"), prop_name='owner_name'),
+            DataTablesColumn(_("Case Coordinates"), prop_name='coordinates'),
+            DataTablesColumn(_("Link"), prop_name='link'),
+        )
 
     @property
     def template_context(self):
         context = super().template_context
-        context['case_row_order'] = {val.html: idx for idx, val in enumerate(self.headers)}
+        context['case_row_order'] = {column.prop_name: index for index, column in enumerate(self.headers)}
```

Comment:
Apart from the change to names of variables `idx -> index` and `val -> column`, we are also now using `prop_name` instead of `html`. Property `html` is the translated header which was later mapped as the key for a value when parsing case row by using `case_row_order`

https://github.com/dimagi/commcare-hq/blob/00ebc191f42148876984b1b4bb50146997b3c2bd/corehq/apps/geospatial/static/geospatial/js/case_grouping_map.js#L536-543 

While the value would still stay the index, we are changing the keys to be the variable names, and translate later through javascript so make this cleaner and avoid the hidden magic.